pkg/resources/network_policy.go

```diff
@@ -109,111 +138,260 @@ func ReadNetworkPolicy(d *schema.ResourceData, meta interface{}) error {
-				return err
+				return diag.FromErr(err)
+			}
+		case "ALLOWED_NETWORK_RULE_LIST":
+			var networkRules []NetworkRulesSnowflakeDTO
+			err := json.Unmarshal([]byte(desc.Value), &networkRules)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			networkRulesFullyQualified := make([]string, len(networkRules))
+			for i, ele := range networkRules {
+				networkRulesFullyQualified[i] = sdk.NewSchemaObjectIdentifierFromFullyQualifiedName(ele.FullyQualifiedRuleName).FullyQualifiedName()
+			}
+
+			if err = d.Set("allowed_network_rule_list", networkRulesFullyQualified); err != nil {
+				return diag.FromErr(err)
+			}
+		case "BLOCKED_NETWORK_RULE_LIST":
+			var networkRules []NetworkRulesSnowflakeDTO
+			err := json.Unmarshal([]byte(desc.Value), &networkRules)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			networkRulesFullyQualified := make([]string, len(networkRules))
+			for i, ele := range networkRules {
+				networkRulesFullyQualified[i] = sdk.NewSchemaObjectIdentifierFromFullyQualifiedName(ele.FullyQualifiedRuleName).FullyQualifiedName()
+			}
+
+			if err = d.Set("blocked_network_rule_list", networkRulesFullyQualified); err != nil {
+				return diag.FromErr(err)
+
 			}
 		}
 	}
 
-	return err
+	return diags
 }
 
-// UpdateNetworkPolicy implements schema.UpdateFunc.
-func UpdateNetworkPolicy(d *schema.ResourceData, meta interface{}) error {
+func UpdateContextNetworkPolicy(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	name := d.Id()
 	client := meta.(*provider.Context).Client
-	ctx := context.Background()
-	baseReq := sdk.NewAlterNetworkPolicyRequest(sdk.NewAccountObjectIdentifier(name))
 
 	if d.HasChange("comment") {
-		comment := d.Get("comment")
+		comment := d.Get("comment").(string)
+		baseReq := sdk.NewAlterNetworkPolicyRequest(sdk.NewAccountObjectIdentifier(name))
 
-		if c := comment.(string); c == "" {
+		if comment == "" {
 			unsetReq := sdk.NewNetworkPolicyUnsetRequest().WithComment(sdk.Bool(true))
 			err := client.NetworkPolicies.Alter(ctx, baseReq.WithUnset(unsetReq))
 			if err != nil {
-				return fmt.Errorf("error unsetting comment for network policy %v err = %w", name, err)
+				return getUpdateContextDiag("unsetting comment", name, err)
 			}
 		} else {
-			setReq := sdk.NewNetworkPolicySetRequest().WithComment(sdk.String(comment.(string)))
+			setReq := sdk.NewNetworkPolicySetRequest().WithComment(sdk.String(comment))
 			err := client.NetworkPolicies.Alter(ctx, baseReq.WithSet(setReq))
 			if err != nil {
-				return fmt.Errorf("error updating comment for network policy %v err = %w", name, err)
+				return getUpdateContextDiag("updating comment", name, err)
+			}
+		}
+	}
+
+	if d.HasChange("allowed_network_rule_list") {
+		oldList, newList := d.GetChange("allowed_network_rule_list")
+		addedNetworkRuleIdentifiers, removedNetworkRuleIdentifiers := getAddedAndRemovedIdentifiers(oldList, newList)
```

Comment:
Why not just use SET with newList?

Reply:
Because I have overlooked this option and concentrated on add/remove network rule from list. Thanks for the hint. Fixed.